e2e-tests/.ci/server.generate.sh

```diff
@@ -0,0 +1,374 @@
+        service: openldap'
+  fi)
+
+$(if mme2e_is_token_in_list "elasticsearch" "$ENABLED_DOCKER_SERVICES"; then
+    if [ "$MME2E_ARCHTYPE" = "arm64" ]; then
+      echo '
+  elasticsearch:
+    image: mattermostdevelopment/mattermost-elasticsearch:7.17.10
+    platform: linux/arm64/v8
+    restart: "no"
+    network_mode: host
+    extends:
+        file: ../../server/build/gitlab-dc.common.yml
+        service: elasticsearch'
+    else
+      echo '
+  elasticsearch:
+    restart: "no"
+    network_mode: host
+    extends:
+        file: ../../server/build/gitlab-dc.common.yml
+        service: elasticsearch'
+    fi
+  fi)
+
+$(if mme2e_is_token_in_list "keycloak" "$ENABLED_DOCKER_SERVICES"; then
+    echo '
+  keycloak:
+    restart: "no"
+    network_mode: host
+    extends:
+        file: ../../server/build/gitlab-dc.common.yml
+        service: keycloak'
+  fi)
+
+$(if mme2e_is_token_in_list "cypress" "$ENABLED_DOCKER_SERVICES"; then
+    echo '
+  cypress:
+    image: "cypress/browsers:node-18.16.1-chrome-114.0.5735.133-1-ff-114.0.2-edge-114.0.1823.51-1"
+    ### Temporarily disabling this image, until both the amd64 and arm64 version are mirrored
+    # image: "mattermostdevelopment/mirrored-cypress-browsers-public:node-18.16.1-chrome-114.0.5735.133-1-ff-114.0.2-edge-114.0.1823.51-1"
+    entrypoint: ["/bin/bash", "-c"]
+    command: ["until [ -f /var/run/mm_terminate ]; do sleep 5; done"]
+    env_file:
+      - "../../e2e-tests/.ci/.env.dashboard"
+      - "../../e2e-tests/.ci/.env.cypress"
+    environment:
+      REPO: "mattermost"
+      # Cypress configuration
+      HEADLESS: "true"
+      CYPRESS_baseUrl: "http://localhost:8065"
+      CYPRESS_dbConnection: "postgres://mmuser:mostest@localhost:5432/mattermost_test?sslmode=disable&connect_timeout=10"
+      CYPRESS_smtpUrl: "http://localhost:9001"
+      CYPRESS_webhookBaseUrl: "http://localhost:3000"
+      CYPRESS_chromeWebSecurity: "false"
+      CYPRESS_firstTest: "true"
+      CYPRESS_resetBeforeTest: "true"
+      CYPRESS_allowedUntrustedInternalConnections: "localhost webhook-interactions"
+      TM4J_ENABLE: "false"
+      # disable shared memory X11 affecting Cypress v4 and Chrome
+      # https://github.com/cypress-io/cypress-docker-images/issues/270
+      QT_X11_NO_MITSHM: "1"
+      _X11_NO_MITSHM: "1"
+      _MITSHM: "0"
+      # avoid too many progress messages
+      # https://github.com/cypress-io/cypress/issues/1243
+      CI: "1"
+      # Ensure we are independent from the global node environment
+      PATH: /cypress/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+    ulimits:
+      nofile:
+        soft: 8096
+        hard: 1048576
+    working_dir: /cypress
+    network_mode: host
+    volumes:
+      - "../../e2e-tests/cypress/:/cypress"'
+  fi)
+
+$(if mme2e_is_token_in_list "webhook-interactions" "$ENABLED_DOCKER_SERVICES"; then
+    # shellcheck disable=SC2016
+    echo '
+  webhook-interactions:
```

Comment:
I haven't had a chance to check fully but I'm still seeing exited container for `webhook-interactions` with the following error:
```
2023-11-06 22:14:08 npm ERR! code EROFS
2023-11-06 22:14:08 npm ERR! syscall rename
2023-11-06 22:14:08 npm ERR! path /cypress/node_modules/@ampproject/remapping
2023-11-06 22:14:08 npm ERR! dest /cypress/node_modules/@ampproject/.remapping-l06oPECg
2023-11-06 22:14:08 npm ERR! errno -30
2023-11-06 22:14:08 npm ERR! rofs EROFS: read-only file system, rename '/cypress/node_modules/@ampproject/remapping' -> '/cypress/node_modules/@ampproject/.remapping-l06oPECg'
2023-11-06 22:14:08 npm ERR! rofs Often virtualized file systems, or other file systems
2023-11-06 22:14:08 npm ERR! rofs that don't support symlinks, give this error.
2023-11-06 22:14:08 
2023-11-06 22:14:08 npm ERR! A complete log of this run can be found in: /root/.npm/_logs/2023-11-06T14_14_01_412Z-debug-0.log
```

Reply:
Interesting, I cannot reproduce it locally :thinking: perhaps it's MacOS-only? We can take a look at this in a separate PR.

btw it's expected for `webhook-interactions` to fail initially if the `cypress/node_modules` is not there yet (I avoid using plain `npm i` in the webhook-interactions container, as it would compete with the cypress container on that same directory). But I added restarts to it to get around this, and anyway it shouldn't fail with the error message that you posted in such a case.